aristanetworks/ansible-cvp
# Fix(cv_container_v3): When a container does not have configlets, has_configlets() returns True

## Change Summary

A conditional test currently always return True in the method cv_container_tools.ContainerInput.has_configlets() and causes unnecessary API calls and a wrong report to the user.

## Related Issue(s)

Fixes #427

## Component(s) name

`pytest`
`arista.cvp.cv_container_v3`

## Proposed changes

The combination of testing if the value is None and calling the ContainerInput.__normalize() method in the ContainerInput constructor causes the issue.
Also test if the value is an empty list in the ContainerInput.has_configlets() method fixes the issue.
Removing ContainerInput.__normalize() would be good as it is unnecessary due to dict key presence checks in the code.

## How to test

`make test TESTS=unit/test_cv_container_tools.py`

> Use `make test TESTS=unit/test_cv_container_tools.py CLI_LOGGING=DEBUG` if you want to see calls to Mock methods.

## Checklist

### User Checklist


- Write a mock class to mimic CVP database
- Write unit tests for build_topology()
- Implement fix
- Sanitize code

### Repository Checklist
